docs/rfcs/004-multi-file-diff.md

```diff
@@ -0,0 +1,65 @@
+# Multi-file Diffs
+
+## Status
+
+Proposed
+
+## Summary
+
+Users can select multiple files in unstaged changes and staged changes pane and see the diffs of selected files in one view, akin to the [`Files changed` tab in pull requests on github.com](https://github.com/atom/github/pull/1753/files).
+
+## Motivation
+
+So that users can view a full set of changes with more context before staging or committing those changes.
+
+The ability to display multiple diffs in one view will also serve as a building block for the following planned features:
+- [commit pane item](#1655) where it shows all changes in a single commit
+- [new PR review flow](https://github.com/atom/github/blob/master/docs/rfcs/003-pull-request-review.md) that shows all changed files proposed in a PR
+
+
+## Explanation
+
+#### Mock-ups coming soon
+
+
+#### Unstaged Changes pane
+- User can `cmd+click` and select multiple files from the list of unstaged changes, and the pane on the left (see multi-file diff section below) will show diffs of the selected files. That pane will continue to reflect any further selecting/unselecting on the Unstaged Changes pane.
+- Once there is at least one file selected, `Stage All` button should be worded as `Stage Selected`.
```

Comment:
> Once there is at least one file selected, `Stage All` button should be worded as `Stage Selected`.

Personally I think I prefer the current functionality where the `Stage All` button remains, regardless of how many changed files are selected. It's kind of like in the diff view where there's a `Stage File` button at the top. It's a quick way to stage everything regardless of what's selected. I could possibly be convinced otherwise though, especially if there's UXR that shows that users would expect the button to change.

Users can currently stage selected files by hitting enter or right-clicking and selecting `Stage`. Admittedly these are less discoverable than a button that says `Stage Selected`

Reply:
I'm 👍 for leaving the "Stage All" behavior as it is, I think. At the very least we can revisit independently from this RFC to keep our scope under control :wink: